File: BL/src/main/java/com/eighteen/muricabl/DAO/_Interfaces/IUserDAO.java
package com.eighteen.muricabl.DAO._Interfaces;

import com.eighteen.muricabl.DTO._Interfaces.IUser;

import java.util.List;

/**
 * Data access object for the user model.
 */
public interface IUserDAO {
    List<IUser> getAll();
}

File: murica_bl/DAO/_Interfaces/IUserDao.php
<?php

namespace murica_bl\DAO\_Interfaces;

use murica_bl\DTO\_Interfaces\IUser;

interface IUserDao
{
    public function findAll(): array;
    public function findByCrit(IUser $model): IUser;
    public function insert(IUser $model): IUser;
    public function remove(IUser $model): void;
}